src/utils/makeBuildQueryFn.ts

```diff
@@ -1,27 +1,15 @@
 import axios, { AxiosRequestConfig } from 'axios';
 import { stringify } from 'qs';
 
+import { isAbsoluteURL, pick } from './helpers';
+import logger from './logger';
 import PromisifiedSemaphore from './PromisifiedSemaphore';
 
-function pick<T, U extends keyof T>(obj: T, keys: U[]) {
-  const newObj: Partial<T> = {};
-  keys.forEach(key => {
-    if (key in obj) {
-      newObj[key] = obj[key];
-    }
-  });
-
-  return newObj as Pick<T, U>;
-}
-
-function isAbsoluteURL(url) {
-  // A URL is considered absolute if it begins with "<scheme>://" or "//" (protocol-relative URL).
-  // RFC 3986 defines scheme name as a sequence of characters beginning with a letter and followed
-  // by any combination of letters, digits, plus, period, or hyphen.
-  return /^([a-z][a-z\d\+\-\.]*:)?\/\//i.test(url);
-}
-
-export default function makeBuildQueryFn<P, R>(baseUrl: string, maxConcurrentQueries?: number) {
+export default function makeBuildQueryFn<P extends object, R>(
```

Comment:
```suggestion
export function makeBuildQueryFn<P extends object, R>(
```

Reply:
Let's use named exports

Reply:
It isn't a tradition in this project but OK